packages/ui-components/scripts/generate-illustrations.js

```diff
@@ -0,0 +1,105 @@
+const path = require('path');
+const fg = require('fast-glob');
+const fs = require('fs-extra');
+const SVGO = require('svgo');
+const { camelCase, startCase, upperFirst, groupBy, map, find } = require('lodash');
+const Handlebars = require('handlebars'); // using Handlebars for consistency with components-sketch
+
+const { parse } = require('svg-parser');
+const toHTML = require('hast-util-to-html');
+
+const svgo = new SVGO({
+	plugins: [{ removeViewBox: false }, { removeXMLNS: true }]
+});
+
+const INPUT_GLOB = './src/illustrations/**/*.svg';
+const OUTPUT_PATH = './src/components/illustrations';
+const ILLUSTRATION_TEMPLATE_PATH = './scripts/illustration-component.hbs';
+const EXT = '.svg';
+
+main();
+
+async function main() {
+	/* Get SVG data from source files */
+
+	const entries = await fg(INPUT_GLOB);
+
+	/* Read file, optimized svg, extract "state" (default) */
+
+	const files = await Promise.all(
+		entries.map(async filepath => {
+			const file = await fs.readFile(filepath, { encoding: 'utf-8' });
+			const item = await svgo.optimize(file);
+
+			const state = path.basename(filepath, EXT);
+			const cleanPath = path.dirname(filepath).replace(INPUT_GLOB.replace('**/*.svg', ''), '');
+			const pathParts = cleanPath.split('/');
+			const key = 'kv-' + cleanPath.replace(/\//gi, '-');
+			const category = pathParts[0];
+			const name = 'kv-' + pathParts[pathParts.length - 1];
+
+			return {
+				...item,
+				filepath,
+				key,
+				category,
+				name,
+				state
+			};
+		})
+	);
+
+	/* Parse, define names and markup */
+
+	const groupedByState = groupBy(files, 'key');
+
+	const components = map(groupedByState, (item, key) => {
+		const defaultItem = find(item);
+		const tagName = key;
+		const className = upperFirst(camelCase(key));
+
+		return {
+			key,
+			tagName,
+			className,
+			illustration: toHTML(getRoot(parse(defaultItem.data))),
+			viewBox: getViewBox(parse(defaultItem.data))
+		};
+	});
+
+	// HAST https://github.com/syntax-tree/hast
+	function getViewBox(tree) {
+		return tree.children[0].properties.viewBox || '0 0 24 24';
+	}
+	function getRoot(tree) {
+		// skip <svg> root
+		const { children } = tree.children[0];
+
+		return {
+			type: 'root',
+			children: children
+		};
+	}
+
+	/* Get component templates */
+
+	const illustrationTemplateText = fs.readFileSync(ILLUSTRATION_TEMPLATE_PATH, {
```

Comment:
Can we change this to `rawIllustrationTemplate`? I believe this would make it a bit clearer as to what data type we're dealing with between these 2 lines where you read the file and compile the handlebars template.

Reply:
Agree :muscle: